edb/graphql/translator.py

```diff
@@ -108,6 +123,23 @@ class TranspiledOperation(NamedTuple):
     variables_desc: dict
 
 
+class BookkeepDict(dict):
+
+    def __init__(self, values):
+        self.update(values)
+        self.touched = set()
+
+    def __getitem__(self, key):
+        self.touched.add(key)
+        return super().__getitem__(key)
+
+    def values(self):
+        raise NotImplementedError()
+
+    def items(self):
+        raise NotImplementedError()
```

Comment:
I'd add `keys()` / `__contains__()` too to make sure one doesn't accidentally lose tracking when debugging/writing more code.

Reply:
I've put everything that involves values. Listing keys does seem harmless. So kinda, graphql code is expected to never do that. And if we debugging by listing somewhere inside the executor, it should be fine.